cmake-edit-core/src/main/java/ru/urururu/cmakeedit/core/checker/FunctionSimulator.java

```diff
@@ -0,0 +1,67 @@
+package ru.urururu.cmakeedit.core.checker;
+
+import ru.urururu.cmakeedit.core.CommandInvocationNode;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by okutane on 22/08/16.
+ */
+public class FunctionSimulator implements AbstractSimulator.CommandSimulator {
+    private final AbstractSimulator simulator;
+    private final List<CommandInvocationNode> body;
+    private Stage stage = Stage.Unused;
+    private SimulationState exitState;
+
+    public FunctionSimulator(AbstractSimulator simulator, List<CommandInvocationNode> body) {
+        this.simulator = simulator;
+        this.body = body;
+    }
+
+    @Override
+    public SimulationState simulate(CheckContext ctx, SimulationState state, CommandInvocationNode command) throws LogicalException {
+        switch (stage) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1301)